official/vision/beta/losses/yolo_losses.py

```diff
@@ -0,0 +1,116 @@
+               name='yolo_loss'):
+    """Initializes `YOLOv4` loss. Consists of ciou, giou and diou.
+      - Generalised IoU (considers shape, orientation in addition to overlap)
+      - Center IoU (considers overlap, distance between center, aspect ratio)
+      - Distance IoU (considers distance of center of bbox)
+
+    Args:
+      input_size: `int`, Dimension of input image
+      num_classes: `int`, Number of foreground classes.
+      iou_loss_thres: `float`, IoU threshold.
+      reduction: (Optional) Type of `tf.keras.losses.Reduction` to apply to
+        loss. Default value is `AUTO`. `AUTO` indicates that the reduction
+        option will be determined by the usage context. For almost all cases
+        this defaults to `SUM_OVER_BATCH_SIZE`. When used with
+        `tf.distribute.Strategy`, outside of built-in training loops such as
+        `tf.keras` `compile` and `fit`, using `AUTO` or `SUM_OVER_BATCH_SIZE`
+        will raise an error. Please see this custom training [tutorial](
+          https://www.tensorflow.org/tutorials/distribute/custom_training) for
+            more details.
+      name: Optional name for the op. Defaults to 'yolo_loss'.
+    """
+    self.input_size = tf.cast(input_size, tf.float32)
+    self.num_classes = num_classes
+    self.iou_loss_thres = iou_loss_thres
+    super(YoloLoss, self).__init__(reduction=reduction, name=name)
+
+  def __call__(self, 
+               pred: tf.Tensor, 
+               conv: tf.Tensor, 
+               label: tf.Tensor, 
+               bboxes: tf.Tensor):
+    """Invokes the `YoloLoss`.
+
+    Args:
+      pred: `tf.Tensor` of shape [batch, height, width, num_anchors, 5 + classes]
+        denoting actual predictions for each scale
+      conv: `tf.Tensor` of shape [batch, height, width, num_anchors, 5 + classes]
+        denoting raw logits from final convolution
+      label: `tf.Tensor` of shape [batch, height, width, num_anchors, 5 + classes]
+        denoting groundtruth labels scaled according to feature size
+      bboxes: `tf.Tensor` of shape [batch, height, width, num_anchors, 5 + classes]
+        denoting ground truth bounding boxes scaled according to feature size
+
+    Returns:
+      giou_loss: Summed loss float `Tensor`,
+      conf_loss: Summed loss float `Tensor`,
+      prob_loss: Summed loss float `Tensor`
+    """
+    conv_shape  = tf.shape(conv)
+    batch_size  = conv_shape[0]
+    output_size = conv_shape[1]
+    conv = tf.reshape(conv, (batch_size, output_size, output_size, 3, 5 + self.num_classes))
+
+    conv_raw_conf = conv[:, :, :, :, 4:5]
+    conv_raw_prob = conv[:, :, :, :, 5:]
+
+    pred_xywh     = pred[:, :, :, :, 0:4]
+    pred_conf     = pred[:, :, :, :, 4:5]
+
+    label_xywh    = label[:, :, :, :, 0:4]
+    respond_bbox  = label[:, :, :, :, 4:5]
+    label_prob    = label[:, :, :, :, 5:]
+
+    giou = tf.expand_dims(box_ops.compute_giou(pred_xywh, label_xywh)[1], axis=-1)
```

Comment:
what does giou stand for?

Reply:
generalized iou https://github.com/whizzmobility/models/blob/47c0a574be1a6b830914375237e0ff6ded7a03a8/official/vision/beta/losses/yolo_losses.py#L22